app/src/main/java/com/isc/hermes/controller/IncidentFormController.java

```diff
@@ -105,19 +126,71 @@ private void handleUploadResponse(Integer responseCode) {
-        Spinner incidentType = ((AppCompatActivity) context).findViewById(R.id.incident_spinner);
-        ArrayAdapter<CharSequence> adapter=ArrayAdapter.createFromResource(context, R.array.incidents_type, R.layout.incident_spinner_items);
-        adapter.setDropDownViewResource(android.R.layout.simple_spinner_item);
-        incidentType.setAdapter(adapter);
+        String[] incidentTypes = context.getResources().getStringArray(R.array.incidents_type);
+        String[] incidentTypeColors = context.getResources().getStringArray(R.array.incidents_type_colors);
+        String[] incidentTypeIcons = context.getResources().getStringArray(R.array.incidents_type_icons);
+
+        if (incidentTypes.length == incidentTypeColors.length &&
+                incidentTypeIcons.length == incidentTypes.length) {
+            for (int i = 0; i < incidentTypes.length; i++) {
+                IncidentTypeButton incidentTypeButton = new IncidentTypeButton();
+                Button button = incidentTypeButton.getIncidentTypeButton(
+                        context,
+                        incidentTypes[i].toLowerCase(),
+                        Color.parseColor((String) incidentTypeColors[i]),
+                        incidentTypeIcons[i]);
+                setIncidentButtonAction(button);
+
+                incidentTypesContainer.addView(button);
+            }
+        } else {
+            Timber.i("The arrays about colors, icons and the type buttons have a different size");
+        }
+
+        setEstimatedTimePicker();
+    }
 
+    /**
+     * This method set the content of the time picker,
+     * and set the number picker according the kind of time.
+     */
+    private void setEstimatedTimePicker() {
         Spinner incidentEstimatedTime = ((AppCompatActivity) context).findViewById(R.id.incident_time_spinner);
         ArrayAdapter<CharSequence> adapterTime=ArrayAdapter.createFromResource(context, R.array.incidents_estimated_time, R.layout.incident_spinner_items);
         adapterTime.setDropDownViewResource(android.R.layout.simple_spinner_item);
-        incidentEstimatedTime.setAdapter(adapterTime);
 
+        incidentEstimatedTime.setAdapter(adapterTime);
         NumberPicker incidentTimePicker = ((AppCompatActivity) context).findViewById(R.id.numberPicker);
-        incidentTimePicker.setMinValue(1);
-        incidentTimePicker.setMaxValue(10);
+        incidentTimePicker.setDescendantFocusability(NumberPicker.FOCUS_BLOCK_DESCENDANTS);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            incidentTimePicker.setTextColor(Color.BLACK);
+        }
+        incidentEstimatedTime.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                String selectedOption = adapterTime.getItem(position).toString();
+
+                if (selectedOption.equals("min")) {
```

Comment:
You may choose to use a Switch to avoid repeating else if

Reply:
Done